scriptworker/utils.py

```diff
@@ -306,3 +314,19 @@ def format_json(data):
         str: the formatted json.
     """
     return json.dumps(data, indent=2, sort_keys=True)
+
+
+async def download_file(context, url, abs_filename, chunk_size=128):
+    log.info("Downloading %s", url)
+    parent_dir = os.path.dirname(abs_filename)
+    async with context.session.get(url) as resp:
```

Comment:
considering removing `context` from the arg list, and having an optional kwarg `session`, overrideable from `download_artifacts`.  That would help avoid the weird context.session swapping.